openvalidators/neuron.py

```diff
@@ -131,22 +133,31 @@ def __init__(self):
         if self.config.neuron.mock_reward_models:
             self.reward_functions = []
             self.masking_functions = [
-                MockRewardModel('blacklist_filter'), 
-                MockRewardModel('nsfw_filter') 
+                MockRewardModel(RewardModelType.blacklist.value),
+                MockRewardModel(RewardModelType.nsfw.value),
             ]
             bt.logging.debug(str(self.reward_functions))
         else:
-            self.reward_functions = [ 
-                OpenAssistantRewardModel( device = self.device ) if not self.config.neuron.openassistant_off else MockRewardModel('rlhf_reward_model'), 
-                ReciprocateRewardModel( device = self.device ) if not self.config.neuron.reciprocate_off else MockRewardModel('reciprocate_reward_model'),
-                DahoasRewardModel( path = self.config.neuron.full_path, device = self.device ) if not self.config.neuron.dahoas_off else MockRewardModel('dahoas_reward_model'),
-                DiversityRewardModel( device = self.device ) if not self.config.neuron.diversity_off else MockRewardModel('diversity_reward_model'),
-                PromptRewardModel( device = self.device ) if not self.config.neuron.prompt_based_off else MockRewardModel('prompt_reward_model'),
+            reward_config = RewardFrameworkConfig(config_file_path=self.config.neuron.reward_config_path)
+
+            self.reward_functions = [
+                OpenAssistantRewardModel(device=self.device, model_weight=reward_config.rlhf_model_weight)
+                    if not self.config.neuron.openassistant_off else MockRewardModel(RewardModelType.rlhf.value),
```

Comment:
A potential problem with having weight separate from the off flag is that if the weight is 0 then the model should ideally be off, despite what the flag specifies.
It may be more natural to just specify a weight config and determine if the model should be off/on.
See: https://github.com/opentensor/validators/compare/sequence_rewards...feature/reward-stack-weights?expand=1

Reply:
The implicit rule of `--model-flag=0` == **do not start the model** seems less natural to me than `--model-flag=on/off` where the configs are isolated into a centralized point of focus.

The rationale behind this implementation is that having multiple approaches (file and flags) adds to the point of failures and increases complexity, so in the choice on one of them, I thought that a config file would be more transparent than scattered flags throughout the code.

But considering your other comments I think we can take a different route on this one, so following your idea, will push some other design